unit/java_bytecode/java_bytecode_parser/java_bytecode_parse_lambda_method_table.cpp

```diff
@@ -61,180 +61,212 @@ SCENARIO(
           "returns and the method it references should have an appropriate "
           "descriptor")
         {
+          const std::string lambda_method_ref =
+            compiler == "eclipse" ? "StaticLambdas.lambda$0:()V"
+                                  : "StaticLambdas.lambda$static$0:()V";
+          const std::string method_type = "()V";
           const lambda_method_handlet &lambda_entry =
             require_parse_tree::require_lambda_entry_for_descriptor(
-              parsed_class, "()V");
+              parsed_class, lambda_method_ref, method_type);
 
           const irep_idt &lambda_impl_name = lambda_entry.lambda_method_name;
 
           const auto lambda_method =
             require_parse_tree::require_method(parsed_class, lambda_impl_name);
-          REQUIRE(id2string(lambda_method.descriptor) == "()V");
+          REQUIRE(id2string(lambda_method.descriptor) == method_type);
         }
 
         // Parameter lambdas
         THEN(
           "There should be an entry for the lambda that takes parameters and "
-          "the "
-          "method it references should have an appropriate descriptor")
+          "the method it references should have an appropriate descriptor")
         {
-          std::string descriptor = "(ILjava/lang/Object;LDummyGeneric;)V";
+          const std::string lambda_method_ref =
+            compiler == "eclipse"
+              ? "StaticLambdas.lambda$1:(ILjava/lang/Object;LDummyGeneric;)V"
+              : "StaticLambdas.lambda$static$1:(ILjava/lang/"
+                "Object;LDummyGeneric;)V";
+          const std::string method_type =
```

Comment:
If you pull the string up a line you can suffix it to both method name variants to reduce duplication/stop the lines wrapping making it harder to follow what is going on:

```
const std::string lambda_method_ref =
  compiler == "eclipse"
    ? "StaticLambdas.lambda$1:" + method_type
    : "StaticLambdas.lambda$static$1:" + method_type;
```

Reply:
Also why change from `descriptor` to `method_type`, is descriptor not more precise?

Reply:
The names for both variables come from the names of the fields of `lambda_method_handlet` which (mostly) come from the bytecode specification. The thing with using `method_type` in declaration of `lambda_method_ref` would only work if the lambda does not capture anything so it would only work for some of the examples in the tests. I would prefer to keep it consistent.

Reply:
This is also the reason why I introduced the `lambda_method_ref` as a field, because it is unambiguous unlike `method_type`.